src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -81,6 +81,8 @@ internal sealed class HttpConnectionPool : IDisposable
         /// <summary>Options specialized and cached for this pool and its key.</summary>
         private readonly SslClientAuthenticationOptions? _sslOptionsHttp11;
         private readonly SslClientAuthenticationOptions? _sslOptionsHttp2;
+        // ToDo: can we forgo ALPN completely if we now that downgrade is forbiden by VersionPolicy?
+        private readonly SslClientAuthenticationOptions? _sslOptionsHttp2Only;
```

Comment:
We should always use ALPN when we're using TLS.

Reply:
advertising preference/requirements can help with troubleshooting.